static/js/ducks/observingRun.js

```diff
@@ -0,0 +1,69 @@
+import messageHandler from 'baselayer/MessageHandler';
+
+import * as API from '../API';
+import store from '../store';
+import { REFRESH_SOURCE } from "./source";
+
+export const FETCH_OBSERVING_RUN = 'skyportal/FETCH_OBSERVING_RUN';
+export const FETCH_OBSERVING_RUN_OK = 'skyportal/FETCH_OBSERVING_RUN_OK';
+
+export const FETCH_ASSIGNMENT = 'skyportal/FETCH_ASSIGNMENT';
+export const FETCH_ASSIGNMENT_OK = 'skyportal/FETCH_ASSIGNMENT_OK';
+
+export const fetchObservingRun = (id) => (
+  API.GET(`/api/observing_run/${id}`, FETCH_OBSERVING_RUN)
+);
+
+export const fetchAssignment = (id) => (
+  API.GET(`/api/assignment/${id}`, FETCH_ASSIGNMENT)
+);
+
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch, getState) => {
+  const { observingRun } = getState();
+
+  /* TODO: Normalize state shape to eliminate the action processor below.
```

Comment:
Please file a corresponding tracking issue, if it does not already exist.

Reply:
This is ch1046